### PR TITLE
Request the review of a set of teams

### LINKED ISSUE #26: Assign team reviewer
**Is your feature request related to a problem? Please describe.**
Currently, we can't assign a team to review.

**Describe the solution you'd like**
Develop a builtin `$assignTeamReviewer()` that given a team slug, assigns the team to review the pull request.

**Describe alternatives you've considered**
*None*.

**Additional context**
*None*.

### PR DESCRIPTION
**Pull request by shay2025**
This pull request introduces a new action called $assignTeamReviewer() that requests the review of a given set of team slugs.

It was also added unit tests for this action.

Documentation: https://github.com/reviewpad/docs/pull/3

Closes #26.

### CODE REVIEW

File: plugins/aladino/actions.go [hunk 1]
```diff
@@ -201,6 +201,46 @@ func assignReviewerCode(e aladino.Env, args []aladino.Value) error {
...
+	if !arg.HasKindOf(aladino.ARRAY_VALUE) {
+		return fmt.Errorf("assignTeamReviewer: requires array argument, got %v", arg.Kind())
+	}
+
+	teamReviewers := arg.(*aladino.ArrayValue).Vals
+
+	teamReviewersSlugs := []string{}
+
+	for _, team := range teamReviewers {
+		if !team.HasKindOf(aladino.STRING_VALUE) {
```
Comment (marcelosousa) [inline plugins/aladino/actions.go:226]: This is not necessary as in the type declaration we already specify that it is an array of strings.

File: plugins/aladino/actions.go [hunk 2]
```diff
@@ -201,6 +201,46 @@ func assignReviewerCode(e aladino.Env, args []aladino.Value) error {
...
+	}
+
+	arg := args[0]
+	if !arg.HasKindOf(aladino.ARRAY_VALUE) {
+		return fmt.Errorf("assignTeamReviewer: requires array argument, got %v", arg.Kind())
+	}
+
+	teamReviewers := arg.(*aladino.ArrayValue).Vals
+
+	teamReviewersSlugs := []string{}
```
Comment (marcelosousa) [inline plugins/aladino/actions.go:223]: Use `make([]string, ...)`

File: plugins/aladino/actions.go [hunk 3]
```diff
@@ -201,6 +201,46 @@ func assignReviewerCode(e aladino.Env, args []aladino.Value) error {
...
+	if len(args) < 1 {
+		return fmt.Errorf("assignTeamReviewer: expecting at least 1 argument")
+	}
+
+	arg := args[0]
+	if !arg.HasKindOf(aladino.ARRAY_VALUE) {
+		return fmt.Errorf("assignTeamReviewer: requires array argument, got %v", arg.Kind())
+	}
+
+	teamReviewers := arg.(*aladino.ArrayValue).Vals
```
Comment (marcelosousa) [inline plugins/aladino/actions.go:221]: Please check that `len(teamReviewers) > 0` and return error otherwise

File: plugins/aladino/actions.go [hunk 4]
```diff
@@ -201,6 +201,46 @@ func assignReviewerCode(e aladino.Env, args []aladino.Value) error {
...
+	return &aladino.BuiltInAction{
+		Type: aladino.BuildFunctionType([]aladino.Type{aladino.BuildArrayOfType(aladino.BuildStringType())}, nil),
+		Code: assignTeamReviewerCode,
+	}
+}
+
+func assignTeamReviewerCode(e aladino.Env, args []aladino.Value) error {
+	if len(args) < 1 {
+		return fmt.Errorf("assignTeamReviewer: expecting at least 1 argument")
+	}
```
Comment (marcelosousa) [inline plugins/aladino/actions.go:214]: This is not necessary as in the type declaration we already declare that this built-in must have one argument.

File: plugins/aladino/actions.go [hunk 5]
```diff
@@ -201,6 +201,46 @@ func assignReviewerCode(e aladino.Env, args []aladino.Value) error {
...
+		if !team.HasKindOf(aladino.STRING_VALUE) {
+			return fmt.Errorf("assignTeamReviewer: requires array of strings, got array with value of %v", team.Kind())
+		}
+		
+		teamReviewersSlugs = append(teamReviewersSlugs, team.(*aladino.StringValue).Val)
+	}
+
+	prNum := utils.GetPullRequestNumber(e.GetPullRequest())
+	owner := utils.GetPullRequestOwnerName(e.GetPullRequest())
+	repo := utils.GetPullRequestRepoName(e.GetPullRequest())
```
Comment (marcelosousa) [inline plugins/aladino/actions.go:235]: Nitpick: I think it is more readable if we assign `e.GetPullRequest()` to an auxiliary variable.

File: plugins/aladino/actions_test.go [hunk 6]
```diff
@@ -18,6 +18,76 @@ import (
...
+	mockedEnv, err := mockDefaultEnv()
+	if err != nil {
+		log.Fatalf("mockDefaultEnv failed: %v", err)
+	}
+
+	args := []aladino.Value{aladino.BuildArrayValue([]aladino.Value{aladino.BuildStringValue("team"), aladino.BuildFalseValue()})}
+	err = assignTeamReviewerCode(mockedEnv, args)
+
+	assert.EqualError(t, err, "assignTeamReviewer: requires array of strings, got array with value of BoolValue")
+}
```
Comment (marcelosousa) [inline plugins/aladino/actions_test.go:59]: I think these tests are not necessary because before running built-ins we pass the type checker which will validate these. However, we need to add a unit test that we don't pass an empty array of strings.

File: plugins/aladino/actions_test.go [hunk 7]
```diff
@@ -18,6 +18,76 @@ import (
...
+	)
+	if err != nil {
+		log.Fatalf("mockDefaultEnv failed: %v", err)
+	}
+
+	args := []aladino.Value{aladino.BuildArrayValue([]aladino.Value{aladino.BuildStringValue("core"), aladino.BuildStringValue("reviewpad-project")})}
+	err = assignTeamReviewerCode(mockedEnv, args)
+
+	assert.Nil(t, err)
+	assert.Equal(t, teamsToRequestForReview, requestedTeamReviewers)
```
Comment (marcelosousa) [inline plugins/aladino/actions_test.go:88]: As standard in Go, lets keep the naming of the variables in these tests starting with `want` and `got`. Please rename `teamsToRequestForReview` to `wantTeamReviewers` and `requestedTeamReviewers` to `gotTeamReviewers`.

**Review by marcelosousa — CHANGES_REQUESTED** [review]
Thanks 👍 I left some comments in the code!

File: plugins/aladino/actions.go (hunk 1 above)
Comment (shay2025) [reply plugins/aladino/actions.go:226]: If so, then we don't need this:
https://github.com/reviewpad/reviewpad/blob/1e99e64910bb6d8a7fb3e7eb70683b17e8de6d23/plugins/aladino/actions.go#L128-L130

Comment (marcelosousa) [reply plugins/aladino/actions.go]: That's true. We can do that fix in another pull request :)

File: plugins/aladino/actions_test.go [hunk 8]
```diff
@@ -18,6 +18,52 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+type TeamReviewersRequestPostBody struct {
+	TeamReviewers []string `json:"team_reviewers"`
+}
+
+func TestAssignTeamReviewers_WhenNoTeamSlugsAreProvided(t *testing.T) {
```
Comment (marcelosousa) [inline plugins/aladino/actions_test.go:25]: Since the built-in name is `assignTeamReviewer` shouldn't these tests be called `TestAssignTeamReviewer_XXX`? There's an extra `s` in `Reviewer`.

**Review by marcelosousa — APPROVED** [review]
Just a small comment in the naming of the test functions. Otherwise seems solid - thanks! 🚀

File: plugins/aladino/actions_test.go [hunk 9]
```diff
@@ -18,6 +18,52 @@ import (
...
+	}
+
+	args := []aladino.Value{aladino.BuildArrayValue([]aladino.Value{})}
+	err = assignTeamReviewerCode(mockedEnv, args)
+
+	assert.EqualError(t, err, "assignTeamReviewer: requires at least 1 team to request for review")
+}
+
+func TestAssignTeamReviewers(t *testing.T) {
+	wantTeamReviewers := []string{
```
Comment (ferreiratiago) [inline plugins/aladino/actions_test.go:38]: I will suggest to add `teamA` and `teamB` as variable, i.e.

 ```go
teamA := "core"
teamB := "reviewpad-project"
wantTeamReviewers := []string{teamA, teamB}
```

**Review by ferreiratiago — APPROVED** [review]
Just a minor comment.

**Comment by ferreiratiago** [comment]
Can we add the related `reviewpad/docs` pull request in the pull request description?

**Review by marcelosousa — APPROVED** [review]
